tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks/CopyReferenceOnlyPackages.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.SourceBuild.Tasks
+{
+    /// <summary>
+    /// For each nupkg directory under the PackageCacheDir, find all directories containing only
+    /// dlls under the /ref/ folder.  These are packages that contain references only.
+    /// Copies all found expanded reference-only packages to the destination directory, excluding
+    /// the .nupkg file.  
+    /// </summary>
+    public class CopyReferenceOnlyPackages : Task
+    {
+        /// <summary>
+        /// Package cache dir containing prebuilt nupkgs. Path is expected to be like:
+        /// 
+        /// {PackageCacheDir}/{lowercase id}/{version}/{lowercase id}.{version}.nupkg
+        ///
+        /// This code assumes that these are expanded packages loaded from nuget.
+        /// </summary>
+        [Required]
+        public string PackageCacheDir { get; set; }
+
+        /// <summary>
+        /// The destination directory for the dlls in reference packages.
+        /// Paths are preserved in the destination.
+        /// </summary>
+        [Required]
+        public string DllDestinationDir { get; set; }
+
+        /// <summary>
+        /// The destination directory for the nupkg files identified 
+        /// as containing reference-only packages.
+        /// </summary>
+        [Required]
+        public string IdentifiedPackagesDir { get; set; }
+
+        /// <summary>
+        /// The destination directory for the reference packages.
+        /// </summary>
+        [Required]
+        public string DestinationDir { get; set; }
+
+        public override bool Execute()
+        {
+            DateTime startTime = DateTime.Now;
+
+            var referenceOnlyPackageDirectories = Directory.EnumerateFiles(PackageCacheDir, "*.nupkg", SearchOption.AllDirectories)
```

Comment:
Line length--maybe add a helper method that is always `SearchOption.AllDirectories` to make this more concise throughout. (I also wouldn't object to chopping after `Directory`.)